test/controllers/products_controller_test.rb

```diff
@@ -238,21 +241,27 @@
         end
       end
 
-      describe "update as Guest" do
-        it "should respond with found and error message to prompt user to login" do
+      describe 'update as Guest' do
+        it 'should respond with found and error message to prompt user to login' do
           patch product_path(Product.first.id)
 
           must_respond_with :found
-          expect(flash[:error]).must_equal "You must be logged in to do this action"
+          expect(flash[:error]).must_equal 'You must be logged in to do this action'
           must_respond_with :redirect
           must_redirect_to root_path
         end
       end
     end
   end
 
-  # it "should get retire" do
-  #   get products_retire_url
-  #   value(response).must_be :success?
-  # end
+  describe "retire" do
+
+    it "should change the retired status of the product" do
```

Comment:
Edge cases? (not logged in, non-existant product)

Reply:
Got it! I'll add these during project time.

Reply:
added not logged in and nonexistent product edge cases!